### PR TITLE
python311Packages.shimmy: init at 1.3.0, python311Packages.dm-control: init at 1.0.19, python311Packages.noxe-xunitmp: init at 0.4.1

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Add python package [shimmy](https://github.com/Farama-Foundation/Shimmy), an API conversion tool for popular external reinforcement learning environments.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/dm-control/default.nix
Comment (DontEatOreo) [inline pkgs/development/python-modules/dm-control/default.nix:97]:
```suggestion
  meta = {
    changelog = "https://github.com/google-deepmind/dm_control/releases/tag/${version}";
    description = "Google DeepMind's software stack for physics-based simulation and Reinforcement Learning environments, using MuJoCo";
    homepage = "https://github.com/google-deepmind/dm_control";
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ GaetanLepage ];
```

File: pkgs/development/python-modules/nose-xunitmp/default.nix
Comment (DontEatOreo) [inline pkgs/development/python-modules/nose-xunitmp/default.nix:37]:
```suggestion
  meta = {
    description = "Xunit output when running multiprocess tests using nose";
    homepage = "https://pypi.org/project/nose_xunitmp/";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ GaetanLepage ];
```

File: pkgs/development/python-modules/shimmy/default.nix
Comment (DontEatOreo) [inline pkgs/development/python-modules/shimmy/default.nix:64]:
```suggestion
  meta = {
    changelog = "https://github.com/Farama-Foundation/Shimmy/releases/tag/v${version}";
    description = "An API conversion tool for popular external reinforcement learning environments";
    homepage = "https://github.com/Farama-Foundation/Shimmy";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ GaetanLepage ];
```

File: pkgs/development/python-modules/dm-control/default.nix
Comment (DontEatOreo) [reply pkgs/development/python-modules/dm-control/default.nix:97]: Using `with` is discouraged, you can read about it [here](https://nix.dev/guides/best-practices.html#with-scopes) and [here](https://github.com/NixOS/nixpkgs/pull/293767)

File: pkgs/development/python-modules/dm-control/default.nix
Comment (DontEatOreo) [inline pkgs/development/python-modules/dm-control/default.nix:41]:
```suggestion
    rev = version;
```

https://github.com/NixOS/nixpkgs/blob/65d153261e9f006909d755d4c8e90058f34080c2/CONTRIBUTING.md?plain=1#L722-L736

**Review by 4evy — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/dm-control/default.nix
Comment (GaetanLepage) [reply pkgs/development/python-modules/dm-control/default.nix]: Ok thanks !
This is how `nix-init` does it though, so it might be worth adapting the tool. cc @figsoda

Comment (GaetanLepage) [reply pkgs/development/python-modules/dm-control/default.nix]: This has already been noticed: https://github.com/nix-community/nix-init/issues/414

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309980` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.dm-control</li>
    <li>python311Packages.dm-control.dist</li>
    <li>python311Packages.nose-xunitmp</li>
    <li>python311Packages.nose-xunitmp.dist</li>
    <li>python311Packages.shimmy</li>
    <li>python311Packages.shimmy.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 309980` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.dm-control</li>
    <li>python311Packages.dm-control.dist</li>
    <li>python311Packages.nose-xunitmp</li>
    <li>python311Packages.nose-xunitmp.dist</li>
    <li>python311Packages.shimmy</li>
    <li>python311Packages.shimmy.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/dm-control/default.nix
Comment (drupol) [reply pkgs/development/python-modules/dm-control/default.nix:40]: Waiting for @DontEatOreo before proceeding further. I have no opinion on this. One or the other it's fine for me.

Comment (DontEatOreo) [reply pkgs/development/python-modules/dm-control/default.nix:40]: I think @GaetanLepage raises a valid concern, and actually, I agree with that committer. In that case, you can ignore this change. Also, could you please link to that committer's comment about this?

On another note, maybe there should be a PR that clarifies this in the manual or somewhere else? I'm currently looking at the [manual](https://nixos.org/manual/nixpkgs/unstable/ ) and there are examples of `rev = version;` in many places. [Searching for `"rev = version;"` on nixpkgs reveals it's widely used as well](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+%22rev+%3D+version%3B%22+path%3A%2F%5Epkgs%5C%2F%2F&type=code)

Comment (drupol) [reply pkgs/development/python-modules/dm-control/default.nix:40]: Fair enough, on my side I'll merge this PR once CI is done then.

**Comment by DontEatOreo** [comment]
Could you update the PR title to include all three packages so it's clearer what is being added

File: pkgs/development/python-modules/dm-control/default.nix
Comment (GaetanLepage) [reply pkgs/development/python-modules/dm-control/default.nix:40]: > I think @GaetanLepage raises a valid concern, and actually, I agree with that committer. In that case, you can ignore this change. Also, could you please link to that committer's comment about this?

If I remember well, this was @SuperSandro2000, but I wouldn't be able to find the comment as it was ~1.5 years ago.

> On another note, maybe there should be a PR that clarifies this in the manual or somewhere else? I'm currently looking at the [manual](https://nixos.org/manual/nixpkgs/unstable/) and there are examples of `rev = version;` in many places. [Searching for `"rev = version;"` on nixpkgs reveals it's widely used as well](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+%22rev+%3D+version%3B%22+path%3A%2F%5Epkgs%5C%2F%2F&type=code)

I asked around and no-one seems to have a strong preference.
Relevant work: https://github.com/NixOS/nixpkgs/pull/153386

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 309980` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.dm-control</li>
    <li>python311Packages.dm-control.dist</li>
    <li>python311Packages.nose-xunitmp</li>
    <li>python311Packages.nose-xunitmp.dist</li>
    <li>python311Packages.shimmy</li>
    <li>python311Packages.shimmy.dist</li>
  </ul>
</details>

**Pull request merged**
